geth/transactions/transactor.go

```diff
@@ -0,0 +1,187 @@
+package transactions
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/status-im/status-go/geth/account"
+	"github.com/status-im/status-go/geth/rpc"
+	"github.com/status-im/status-go/sign"
+)
+
+const (
+	// sendTxTimeout defines how many seconds to wait before returning result in sentTransaction().
+	sendTxTimeout  = 300 * time.Second
+	rpcCallTimeout = time.Minute
+
+	defaultGas = 90000
+)
+
+// Transactor validates, signs transactions.
+// It uses upstream to propagate transactions to the Ethereum network.
+type Transactor struct {
+	signRequests   *sign.PendingRequests
+	upstream       *gethTransactor
+	sendTxTimeout  time.Duration
+	rpcCallTimeout time.Duration
+	networkID      uint64
+
+	addrLock   *AddrLocker
+	localNonce sync.Map
+	log        log.Logger
+}
+
+// NewTransactor returns a new Manager.
+func NewTransactor(signRequests *sign.PendingRequests) *Transactor {
+	return &Transactor{
+		signRequests:   signRequests,
+		addrLock:       &AddrLocker{},
+		sendTxTimeout:  sendTxTimeout,
+		rpcCallTimeout: rpcCallTimeout,
+		localNonce:     sync.Map{},
+		log:            log.New("package", "status-go/geth/transactions.Manager"),
+	}
+}
+
+// SetUpstream selects a correct network and RPC client.
+func (t *Transactor) SetUpstream(networkID uint64, rpcClient *rpc.Client) {
```

Comment:
`SetUpstream` is not accurate as `rpcClient` can be only in-proc RPC client. Recommend splitting it into `SetNetworkID` and `SetRPCClient`.

Reply:
For this transactor it is upstream, we forward `eth_sendTransaction` somewhere, that could be an upstream node or our local node. Both are external to this one.